mandible/umm/ummg.py

```diff
@@ -0,0 +1,210 @@
+
+    def get_archive_distribution_information(self) -> List[dict]:
+        return [{
+            "Name": self.get_file_name(),
+            "SizeInBytes": self.get_product_file_size(),
+            "Format": self.get_file_type(),
+        }]
+
+    def get_data_identifiers(self) -> List[dict]:
+        return [{
+            "Identifier": self.get_scene_id(),
+            "IdentifierType": "ProducerGranuleId"
+        }]
+
+    def get_collection_long_name(self) -> str:
+        return self.collection_ref_table["long_name"]
+
+    def get_collection_short_name(self) -> str:
+        return self.collection_ref_table["short_name"]
+
+    def get_scene_id(self) -> str:
+        return self.get_file_name_path().name.split(".", 1)[0]
+
+    def get_file_name(self) -> str:
+        return str(self.get_file_name_path())
+
+    def get_product_url(self) -> str:
+        return self.get_product_file_metadata()["uri"]
+
+    def get_product_s3_uri(self) -> str:
+        return self.get_product_file_metadata()["s3uri"]
+
+    def get_product_file_size(self) -> str:
+        return self.get_product_file_metadata()["size"]
+
+    # Start of getters that handle the big picture
+    def get_additional_attributes(self) -> List[dict]:
+        return []
+
+    def get_collection_reference(self) -> dict:
+        return {
+            "EntryTitle": self.get_collection_long_name()
+        }
+
+    def get_data_granule(self) -> dict:
+        return {
+            "ArchiveAndDistributionInformation": self.get_archive_distribution_information(),
+            "DayNightFlag": "Unspecified",
+            "Identifiers": self.get_data_identifiers(),
+            "ProductionDateTime": self.get_product_creation_time().strftime(UMM_DATETIME_FORMAT),
+        }
+
+    def get_granule_ur(self) -> str:
+        return self.get_file_name_path().name.replace(".", "-")
+
+    def get_metadata_specification(self) -> dict:
+        return {
+            "Name": "UMM-G",
+            "URL": "https://cdn.earthdata.nasa.gov/umm/granule/v1.6.4",
+            "Version": "1.6.4"
+        }
+
+    def get_pge_version(self) -> dict:
+        return {}
+
+    def get_platforms(self) -> List[dict]:
+        return [{
+            "ShortName": self.get_mission()
+        }]
+
+    def get_provider_dates(self) -> List[dict]:
+        return [
+            {"Type": "Insert", "Date": self.get_provider_time().strftime(UMM_DATETIME_FORMAT)},
+            {"Type": "Update", "Date": self.get_provider_time().strftime(UMM_DATETIME_FORMAT)},
+        ]
+
+    def get_related_urls(self) -> List[dict]:
+        return [
+            {
+                "URL": self.get_product_url(),
+                "Type": "GET DATA",
+                "Subtype": "VERTEX"
+            },
+            {
+                "URL": self.get_product_s3_uri(),
+                "Type": "GET DATA VIA DIRECT ACCESS"
+            }
+        ]
+
+    def get_spatial_extent(self) -> dict:
```

Comment:
Especially here having a typed return dict would be useful since there's no example to go off of